lgc/builder/BuilderImpl.cpp

```diff
@@ -503,36 +521,130 @@ static Value *traceNonUniformIndex(Value *nonUniformVal) {
 
   return candidateIndex;
 }
+#endif
 
-// =====================================================================================================================
-// Test whether two instructions are identical
-// or are the same operation on identical operands.
-// @param lhs : First instruction
-// @param rhs : Second instruction
-// @return Result of equally test
-static bool instructionsEqual(Instruction *lhs, Instruction *rhs) {
-  if (lhs->isIdenticalTo(rhs))
-    return true;
-
-  if (!lhs->isSameOperationAs(rhs))
-    return false;
-
-  for (unsigned idx = 0, end = lhs->getNumOperands(); idx != end; ++idx) {
-    Value *lhsVal = lhs->getOperand(idx);
-    Value *rhsVal = rhs->getOperand(idx);
-    if (lhsVal == rhsVal)
-      continue;
-    Instruction *lhsInst = dyn_cast<Instruction>(lhsVal);
-    Instruction *rhsInst = dyn_cast<Instruction>(rhsVal);
-    if (!lhsInst || !rhsInst)
-      return false;
-    if (!lhsInst->isIdenticalTo(rhsInst))
-      return false;
+Value *BuilderImpl::emitWaterfallBegin(Instruction *nonUniformInst, ArrayRef<unsigned> operandIdxs,
```

Comment:
Can you add a comment to this method, describing its functionality and the arguments?

Reply:
Done.